src/protocol/sort/bit_permutation.rs

```diff
@@ -33,13 +32,15 @@ use futures::future::try_join_all;
 ///
 /// ## Errors
 /// It will propagate errors from multiplication protocol.
-pub async fn bit_permutation<'a, F: Field, S: SecretSharing<F>>(
-    ctx: ProtocolContext<'a, S, F>,
+pub async fn bit_permutation<
```

Comment:
this is how the protocols would look like if they can work in semi-honest and malicious setting. 

Note that there is no more `SecureMul` constraint because it is a supertrait for  `ProtocolContext`.

Reply:
That's really nice. Thanks for eliminating those constraints!